src/lib.rs

```diff
@@ -589,95 +677,93 @@ impl Contract {
         self.edit_proposal_internal(id, body.into(), proposal.snapshot.labels)
     }
 
-    fn edit_proposal_internal(
-        &mut self,
-        id: ProposalId,
-        body: VersionedProposalBody,
-        labels: HashSet<String>,
-    ) -> Promise {
-        require!(
-            self.is_allowed_to_edit_proposal(id, Option::None),
-            "The account is not allowed to edit this proposal"
-        );
-        let editor_id = env::predecessor_account_id();
-        let mut proposal: Proposal = self
+    #[payable]
+    pub fn edit_proposal_linked_rfp(&mut self, id: ProposalId, rfp_id: Option<RFPId>) -> ProposalId {
+        let proposal: Proposal = self
             .proposals
             .get(id.into())
             .unwrap_or_else(|| panic!("Proposal id {} not found", id))
             .into();
+        let mut body = proposal.snapshot.body.latest_version();
+        body.linked_rfp = rfp_id;
 
-        let proposal_body = body.clone().latest_version();
+        self.edit_proposal_internal(id, body.into(), proposal.snapshot.labels)
+    }
+
+    #[payable]
+    pub fn edit_rfp(
+        &mut self,
+        id: RFPId,
+        body: VersionedRFPBody,
+        labels: HashSet<String>,
+    ) -> RFPId {
+        self.edit_rfp_internal(id, body.clone(), labels)
+    }
 
-        let current_timeline = proposal.snapshot.body.clone().latest_version().timeline;
+    #[payable]
+    pub fn cancel_rfp(&mut self, id: RFPId, proposals_to_cancel: Vec<ProposalId>, proposals_to_unlink: Vec<ProposalId>) -> RFPId {
+        let rfp: RFP = self.get_rfp(id).into();
+        let mut body = rfp.snapshot.body.latest_version();
+        body.timeline = RFPTimelineStatus::Cancelled;
```

Comment:
These are not saved, so you should just remove these two lines. The actual editing happens below in `edit_rfp_timeline` call